code/game/machinery/traitorous_cloner.dm

```diff
@@ -0,0 +1,355 @@
+		say("Initiating scan...")
+
+		spawn(20)
+			clone_occupant(scanner.occupant)
+			loading = FALSE
+			updateUsrDialog()
+			playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
+
+		//No locking an open scanner.
+	else if ((href_list["lock"]) && !isnull(scanner) && scanner.is_operational())
+		if ((!scanner.locked) && (scanner.occupant))
+			scanner.locked = TRUE
+			playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+		else
+			scanner.locked = FALSE
+			playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
+
+	else if (href_list["refresh"])
+		updateUsrDialog()
+		playsound(src, "terminal_type", 25, 0)
+
+	add_fingerprint(usr)
+	updateUsrDialog()
+	return
+
+/obj/machinery/computer/prototype_cloning_traitorous/proc/clone_occupant(occupant)
+	var/mob/living/mob_occupant = get_mob_or_brainmob(occupant)
+	var/datum/dna/dna
+	if(ishuman(mob_occupant))
+		var/mob/living/carbon/C = mob_occupant
+		dna = C.has_dna()
+	if(isbrain(mob_occupant))
+		var/mob/living/brain/B = mob_occupant
+		dna = B.stored_dna
+
+	if(!istype(dna))
+		scantemp = "<font class='bad'>Unable to locate valid genetic data.</font>"
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+		return
+	if((HAS_TRAIT(mob_occupant, TRAIT_HUSK)) && (src.scanner.scan_level < 2))
+		scantemp = "<font class='bad'>Subject's body is too damaged to scan properly.</font>"
+		playsound(src, 'sound/machines/terminal_alert.ogg', 50, 0)
+		return
+	if(HAS_TRAIT(mob_occupant, TRAIT_BADDNA))
+		scantemp = "<font class='bad'>Subject's DNA is damaged beyond any hope of recovery.</font>"
+		playsound(src, 'sound/machines/terminal_alert.ogg', 50, 0)
+		return
+
+	var/clone_species
+	if(dna.species)
+		clone_species = dna.species
+	else
+		var/datum/species/rando_race = pick(GLOB.roundstart_races)
+		clone_species = rando_race.type
+
+	var/obj/machinery/clonepod/pod = GetAvailablePod()
+	//Can't clone without someone to clone.  Or a pod.  Or if the pod is busy. Or full of gibs.
+	if(!LAZYLEN(pods))
+		temp = "<font class='bad'>No Clonepods detected.</font>"
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	else if(!pod)
+		temp = "<font class='bad'>No Clonepods available.</font>"
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	else if(pod.occupant)
+		temp = "<font class='bad'>Cloning cycle already in progress.</font>"
+		playsound(src, 'sound/machines/terminal_prompt_deny.ogg', 50, 0)
+	else
+		pod.growclone(mob_occupant.real_name, dna.uni_identity, dna.mutation_index, null, null, clone_species, dna.features, mob_occupant.faction)
+		temp = "[mob_occupant.real_name] => <font class='good'>Cloning data sent to pod.</font>"
+		playsound(src, 'sound/machines/terminal_prompt_confirm.ogg', 50, 0)
+
+/obj/machinery/computer/prototype_cloning_traitorous/emag_act(emag_user)
+	var/obj/machinery/clonepod/traitorous/P = GetAvailablePod()
+	if(!is_emmaged || !P.is_emmaged())
```

Comment:
Do you really need to check if the pod is emaged? If there is no pod you can emag computer infinitely

Reply:
To me it seems like the best solution other then pass more arguments and overwriting cloning function again, why would you emag a console without a pod in it indefinitely?

Reply:
Won't !P.is_emmaged() be null if there is no pod?

Reply:
Yeah, but why would you hack a console without a cloner? This is kinda pointless so I think it doesn't matter.

Reply:
This will be oversight later since you can spam text with this action